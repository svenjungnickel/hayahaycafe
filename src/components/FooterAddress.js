import React from 'react';
import PropTypes from 'prop-types';
import { graphql, StaticQuery } from 'gatsby';

const AddressItem = ({ item }) => <>{item && <>{item}, </>}</>;

AddressItem.propTypes = {
    item: PropTypes.string.isRequired,
};

const FooterAddress = () => (
    <StaticQuery
        query={graphql`
            query {
                globalSettings: settingsYaml {
                    location {
                        address {
                            company
                            addressLine1
                            addressLine2
                            province
                            postalCode
                            country
                        }
                    }
                }
            }
        `}
        render={({ globalSettings }) => (
            <>
                {!!globalSettings && !!globalSettings.location && !!globalSettings.location.address && (
                    <p>
                        <AddressItem item={globalSettings.location.address.company} />
                        <AddressItem item={globalSettings.location.address.addressLine1} />
                        <AddressItem item={globalSettings.location.address.addressLine2} />
                        <AddressItem item={globalSettings.location.address.province} />
                        <AddressItem item={globalSettings.location.address.postalCode} />
                        {globalSettings.location.address.country && <>{globalSettings.location.address.country}</>}
                    </p>
                )}
            </>
        )}
    />
);

export default FooterAddress;
