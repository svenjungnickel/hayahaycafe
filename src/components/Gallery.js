import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Carousel from 'nuka-carousel';
import FsLightbox from 'fslightbox-react';
import Image from './Image';
import GalleryStyle from '../styles/components/Gallery.module.scss';
import '../styles/libs/fslightbox-react.scss';

const PreviousButton = ({ previousSlide }) => (
    <button className={GalleryStyle.previousButton} aria-label="previous" onClick={previousSlide}>
        &lsaquo;
    </button>
);

PreviousButton.propTypes = {
    previousSlide: PropTypes.func.isRequired,
};

const NextButton = ({ nextSlide }) => (
    <button className={GalleryStyle.nextButton} aria-label="next" onClick={nextSlide}>
        &rsaquo;
    </button>
);

NextButton.propTypes = {
    previousSlide: PropTypes.func.isRequired,
};

const PagingDots = ({ slideCount, slidesToScroll, goToSlide, currentSlide }) => (
    <ul className={GalleryStyle.pagingDots__list}>
        {PagingDots.getIndexes(slideCount, slidesToScroll).map(index => {
            return (
                <li className={GalleryStyle.pagingDots__listItem} key={index}>
                    <button
                        className={GalleryStyle.pagingDots__button}
                        style={{ opacity: currentSlide === index ? 1 : 0.5 }}
                        onClick={goToSlide.bind(null, index)}
                    >
                        &bull;
                    </button>
                </li>
            );
        })}
    </ul>
);

PagingDots.getIndexes = (count, inc) => {
    const arr = [];
    for (let i = 0; i < count; i += inc) {
        arr.push(i);
    }
    return arr;
};

PagingDots.propTypes = {
    slideCount: PropTypes.number.isRequired,
    slidesToScroll: PropTypes.number.isRequired,
    goToSlide: PropTypes.func.isRequired,
    currentSlide: PropTypes.number.isRequired,
};

const Gallery = ({ images }) => {
    const [toggler, setToggler] = useState(false);
    const imageSources = images.map(image => {
        if (!!image.src && !!image.src.childImageSharp) {
            if (image.src.childImageSharp.fluid) {
                return image.src.childImageSharp.fluid.src;
            }

            return image.src.childImageSharp.fixed.src;
        }

        return image.src;
    });

    return (
        <>
            <Carousel
                dragging={true}
                swiping={true}
                wrapAround={true}
                heightMode="first"
                renderCenterLeftControls={props => <PreviousButton {...props} />}
                renderCenterRightControls={props => <NextButton {...props} />}
                renderBottomCenterControls={props => <PagingDots {...props} />}
            >
                {images.map((image, index) => (
                    <Image
                        key={index}
                        src={image.src}
                        alt={image.alt}
                        title={image.title}
                        onClick={() => setToggler(!toggler)}
                    />
                ))}
            </Carousel>
            <FsLightbox toggler={toggler} sources={imageSources} />
        </>
    );
};

Gallery.propTypes = {
    images: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Gallery;
