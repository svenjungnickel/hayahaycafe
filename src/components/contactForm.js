import React, { useState, createRef } from 'react';
import { Alert, Button, Col, Form } from 'react-bootstrap';
import Recaptcha from 'react-google-recaptcha';
import Fade from 'react-reveal/Fade';
import Styles from '../styles/components/contactForm.module.scss';

const RECAPTCHA_KEY = process.env.SITE_RECAPTCHA_KEY;
const CONTACT_FORM_FADE_UP_DURATION = 400;
const SUCCESS_MESSAGE_FADE_IN_DURATION = 200;

function encode(data) {
    return Object.keys(data)
        .map(key => encodeURIComponent(key) + '=' + encodeURIComponent(data[key]))
        .join('&');
}

export default () => {
    const [validated, setValidated] = useState(false);
    const [state, setState] = useState({});
    const [recaptchaError, setRecaptchaError] = useState(false);
    const [success, setSuccess] = useState(false);
    const recaptchaRef = createRef();

    const handleChange = e => {
        setState({ ...state, [e.target.name]: e.target.value });
    };

    const handleRecaptcha = () => {
        setRecaptchaError(false);
    };

    const handleSubmit = event => {
        event.preventDefault();
        event.stopPropagation();
        setValidated(true);

        const recaptchaValue = recaptchaRef.current.getValue() || false;
        if (false === recaptchaValue) {
            setRecaptchaError(true);
            return;
        }

        if (event.currentTarget.checkValidity() === false) {
            return;
        }

        const form = event.target;
        fetch('/', {
            method: 'POST',
            headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
            body: encode({
                'form-name': form.getAttribute('name'),
                'g-recaptcha-response': recaptchaValue,
                ...state,
            }),
        })
            .then(() => {
                console.log('success');
                setSuccess(true);
            })
            .catch(error => alert(error));
    };

    return (
        <div className={Styles.contactForm}>
            <Fade top collapse when={false === success} duration={CONTACT_FORM_FADE_UP_DURATION} unmountOnExit={true}>
                <Form
                    noValidate
                    validated={validated}
                    onSubmit={handleSubmit}
                    name="contact"
                    method="POST"
                    action="/thanks"
                    data-netlify="true"
                    data-netlify-recaptcha="true"
                >
                    <Form.Label>Name *</Form.Label>
                    <Form.Row>
                        <Col xs={12} lg={6}>
                            <Form.Group controlId="contactFirstName">
                                <Form.Control name="firstName" type="text" required onChange={handleChange} />
                                <Form.Text className="text-muted">First name</Form.Text>
                                <Form.Control.Feedback type="invalid">
                                    Please enter your first name.
                                </Form.Control.Feedback>
                            </Form.Group>
                        </Col>
                        <Col xs={12} lg={6}>
                            <Form.Group controlId="contactLastName">
                                <Form.Control name="lastName" type="text" required onChange={handleChange} />
                                <Form.Text className="text-muted">Last name</Form.Text>
                                <Form.Control.Feedback type="invalid">
                                    Please enter your last name.
                                </Form.Control.Feedback>
                            </Form.Group>
                        </Col>
                    </Form.Row>

                    <Form.Group controlId="contactEmail">
                        <Form.Label>Email *</Form.Label>
                        <Form.Control name="email" type="email" required onChange={handleChange} />
                        <Form.Control.Feedback type="invalid">
                            Please enter a valid email address.
                        </Form.Control.Feedback>
                    </Form.Group>

                    <Form.Group controlId="contactSubject">
                        <Form.Label>Subject *</Form.Label>
                        <Form.Control name="subject" type="text" required onChange={handleChange} />
                        <Form.Control.Feedback type="invalid">Please enter a subject.</Form.Control.Feedback>
                    </Form.Group>

                    <Form.Group controlId="contactMessage">
                        <Form.Label>Message *</Form.Label>
                        <textarea
                            name="message"
                            className="form-control"
                            id="contactMessage"
                            rows="5"
                            required
                            onChange={handleChange}
                        />
                        <Form.Control.Feedback type="invalid">Please enter your message.</Form.Control.Feedback>
                    </Form.Group>

                    <Form.Group controlId="contactRecaptcha">
                        <Recaptcha
                            id="contactRecaptcha"
                            ref={recaptchaRef}
                            sitekey={RECAPTCHA_KEY}
                            onChange={handleRecaptcha}
                            className={Styles.contactFormReCaptcha}
                        />
                        {true === recaptchaError && (
                            <div className="invalid-feedback" style={{ display: 'block' }}>
                                Please answer the captcha.
                            </div>
                        )}
                    </Form.Group>

                    <Button variant="primary" type="submit" className={Styles.contactFormSubmitButton}>
                        Submit
                    </Button>
                </Form>
            </Fade>
            <Fade
                collapse
                when={success}
                duration={SUCCESS_MESSAGE_FADE_IN_DURATION}
                delay={CONTACT_FORM_FADE_UP_DURATION}
            >
                <Alert variant="success" className={Styles.contactFormSuccess}>
                    <p>{`Thank you for your message. We're getting in touch with you soon.`}</p>
                </Alert>
            </Fade>
        </div>
    );
};
