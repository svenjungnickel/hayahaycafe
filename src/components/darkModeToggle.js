import React from 'react';
import useDarkMode from 'use-dark-mode';
import ReactSwitch from 'react-switch';

const Switch = props => <ReactSwitch {...props} />;

Switch.defaultProps = {
    checkedIcon: false,
    uncheckedIcon: false,
    height: 24,
    width: 48,
    handleDiameter: 24,
    offColor: `#000`,
    onColor: `#000`,
    boxShadow: `inset 0 0 0 1px #000`,
};

const uncheckedIcon = (
    <div
        style={{
            display: 'flex',
            justifyContent: 'center',
            alignItems: 'center',
            height: '100%',
            fontSize: 15,
            color: 'yellow',
            paddingLeft: 2,
        }}
    >
        ☀
    </div>
);

const checkedIcon = (
    <div
        style={{
            display: 'flex',
            justifyContent: 'center',
            alignItems: 'center',
            height: '100%',
            fontSize: 15,
            color: 'yellow',
            paddingRight: 2,
        }}
    >
        ☾
    </div>
);

const DarkModeToggle = props => {
    const darkMode = useDarkMode(false),
        darkModeValue = null !== darkMode.value ? darkMode.value : false;

    return (
        <Switch
            {...props}
            title="dark mode switch"
            aria-checked={darkModeValue}
            checked={darkModeValue}
            onChange={darkMode.toggle}
            uncheckedIcon={uncheckedIcon}
            checkedIcon={checkedIcon}
        />
    );
};

export default DarkModeToggle;
