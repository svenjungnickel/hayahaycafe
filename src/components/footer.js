import React from 'react';
import { Container, Nav } from 'react-bootstrap';
import SocialMediaIcons from './socialMediaIcons';
import FooterStyles from '../styles/components/footer.module.scss';

export default ({ siteMetadata }) => (
    <footer className={FooterStyles.footer}>
        <div className={FooterStyles.footer__pre}>
            <SocialMediaIcons items={siteMetadata.socialMedia} />
        </div>
        <div className={FooterStyles.footer__main}>
            <Container className="text-center">
                <Nav className="justify-content-center">
                    <Nav.Item>
                        <Nav.Link href="/contact">Contact</Nav.Link>
                    </Nav.Item>
                    <Nav.Item>
                        <Nav.Link href="/legal">Legal</Nav.Link>
                    </Nav.Item>
                    <Nav.Item>
                        <Nav.Link href="/data-privacy">Data Privacy</Nav.Link>
                    </Nav.Item>
                </Nav>

                <div className={FooterStyles.footer__address}>
                    <p>{siteMetadata.address}</p>
                </div>
            </Container>
        </div>
    </footer>
);
