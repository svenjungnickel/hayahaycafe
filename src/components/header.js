import React from 'react';
import Img from 'gatsby-image';
import PropTypes from 'prop-types';
import { Container } from 'react-bootstrap';
import IndexStyles from '../styles/components/header.module.scss';

const Header = ({ header, title, subtitle }) => (
    <header className={IndexStyles.header}>
        <Img fluid={header} alt="header" className={IndexStyles.header__image} />

        <div className={IndexStyles.header__content}>
            <Container>
                <div className={IndexStyles.header__contentInner}>
                    <h1 className="display-1">{title}</h1>
                    <p className="lead">{subtitle}</p>
                </div>
            </Container>
        </div>
    </header>
);

Header.propTypes = {
    header: PropTypes.object.isRequired,
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired,
};

export default Header;
