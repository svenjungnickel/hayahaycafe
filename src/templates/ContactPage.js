import React from 'react';
import PropTypes from 'prop-types';
import { graphql } from 'gatsby';
import { Col, Container, Row } from 'react-bootstrap';
import Layout from '../components/layout';
import Header from '../components/header';
import ContactForm from '../components/contactForm';
import Content, { HTMLContent } from '../components/content';
import Address from '../components/Location';
import Separator from '../components/Separator';
import What3WordsAddress from '../components/What3WordsAddress';
import OpeningHours from '../components/OpeningHours';
import SectionStyles from '../styles/components/section.module.scss';

export const ContactPageTemplate = ({ title, subtitle, headerImage, content, contentComponent }) => {
    const PostContent = contentComponent || Content;

    return (
        <>
            <Header title={title} subtitle={subtitle} headerImage={headerImage} />

            <section className={SectionStyles.section}>
                <Container>
                    <Row>
                        {!!content && (
                            <Col xs={12}>
                                <PostContent content={content} />
                                <Separator />
                            </Col>
                        )}
                        <Col xs={12} sm={4}>
                            <Address />
                            <Separator />
                            <What3WordsAddress />
                            <Separator />
                            <OpeningHours />
                            <Separator className="d-block d-sm-none" />
                        </Col>
                        <Col xs={12} sm={8}>
                            <h2 className="d-block d-sm-none">Contact us</h2>
                            <ContactForm />
                        </Col>
                    </Row>
                </Container>
            </section>
        </>
    );
};

ContactPageTemplate.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired,
    headerImage: PropTypes.oneOfType([PropTypes.object, PropTypes.string]).isRequired,
    content: PropTypes.node,
    contentComponent: PropTypes.func,
};

const ContactPage = ({ data: { page } }) => (
    <Layout meta={page.frontmatter.meta} currentPage="/contact">
        <ContactPageTemplate {...page.frontmatter} content={page.html} contentComponent={HTMLContent} />
    </Layout>
);

ContactPage.propTypes = {
    data: PropTypes.shape({
        markdownRemark: PropTypes.object,
    }).isRequired,
};

export default ContactPage;

export const pageQuery = graphql`
    query ContactPage($id: String!) {
        page: markdownRemark(id: { eq: $id }) {
            html
            frontmatter {
                title
                subtitle
                headerImage {
                    childImageSharp {
                        fluid(maxHeight: 500) {
                            ...GatsbyImageSharpFluid
                        }
                    }
                    publicURL
                }
                meta {
                    title
                    description
                    keywords
                }
            }
        }
    }
`;
