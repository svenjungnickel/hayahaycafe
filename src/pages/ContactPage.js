import React from 'react';
import PropTypes from 'prop-types';
import { graphql } from 'gatsby';
import Layout from '../components/layout';
import SEO from '../components/seo';
import { HTMLContent } from '../components/content';
import ContactPageTemplate from '../templates/ContactPageTemplate';

const ContactPage = ({ data: { page, what3wordsIcon } }) => (
    <Layout currentPage="/contact">
        <SEO
            title={page.frontmatter.title}
            keywords={(page.frontmatter.meta && page.frontmatter.meta.keywords) || []}
        />
        {/*<SEO meta={page.frontmatter.meta || false} title={page.frontmatter.title || false} />*/}

        <ContactPageTemplate
            what3wordsIcon={what3wordsIcon}
            {...page.frontmatter}
            content={page.html}
            contentComponent={HTMLContent}
        />
    </Layout>
);

ContactPage.propTypes = {
    data: PropTypes.shape({
        markdownRemark: PropTypes.object,
        what3wordsIcon: PropTypes.string,
    }),
};

export default ContactPage;

export const pageQuery = graphql`
    query ContactPage($id: String!) {
        page: markdownRemark(id: { eq: $id }) {
            html
            frontmatter {
                title
                subtitle
                what3WordsAddress
                headerImage {
                    childImageSharp {
                        fluid(maxHeight: 500) {
                            ...GatsbyImageSharpFluid
                        }
                    }
                    publicURL
                }
            }
        }
        what3wordsIcon: file(relativePath: { eq: "what3words-icon.png" }) {
            childImageSharp {
                fixed(width: 80, height: 80) {
                    ...GatsbyImageSharpFixed
                }
            }
            publicURL
        }
    }
`;
