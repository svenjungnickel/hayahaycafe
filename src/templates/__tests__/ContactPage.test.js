import React from 'react';
import renderer from 'react-test-renderer';
import ContactPage, { ContactPageTemplate } from '../ContactPage';
import { HTMLContent } from '../../components/Content';
import ContactPageData from '../../__fixtures__/ContactPageData';

jest.mock('../../components/Layout');
jest.mock('../../components/Header');
jest.mock('../../components/ContactForm/ContactForm');
jest.mock('../../components/Separator');

describe('ContactPageTemplate', () => {
    it('renders without content', () => {
        const title = 'title';
        const subtitle = 'subtitle';
        const headerImage = 'headerImage';

        const component = <ContactPageTemplate title={title} subtitle={subtitle} headerImage={headerImage} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders with content', () => {
        const title = 'title';
        const subtitle = 'subtitle';
        const headerImage = 'headerImage';
        const content = 'content';

        const component = (
            <ContactPageTemplate title={title} subtitle={subtitle} headerImage={headerImage} content={content} />
        );
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders with html content', () => {
        const title = 'title';
        const subtitle = 'subtitle';
        const headerImage = 'headerImage';
        const content = <p>HTML content</p>;

        const component = (
            <ContactPageTemplate
                title={title}
                subtitle={subtitle}
                headerImage={headerImage}
                content={content}
                contentComponent={HTMLContent}
            />
        );
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});

describe('ContactPage', () => {
    it('renders correctly', () => {
        const data = {
            page: {
                html: '',
                frontmatter: ContactPageData,
            },
        };

        const component = <ContactPage data={data} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
