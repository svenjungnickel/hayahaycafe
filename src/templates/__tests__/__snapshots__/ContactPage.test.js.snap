// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ContactPage renders correctly 1`] = `
<main>
  <header />
  <section
    className="section"
  >
    <div
      className="container"
    >
      <div
        className="justify-content-md-center row"
      >
        <div
          className="col-md-8 col-12"
        >
          <form />
        </div>
      </div>
    </div>
  </section>
</main>
`;

exports[`ContactPageTemplate renders with content 1`] = `
[
  <header />,
  <section
    className="section"
  >
    <div
      className="container"
    >
      <div
        className="justify-content-md-center row"
      >
        <div
          className="col-md-8 col-12"
        >
          <div>
            content
          </div>
          <hr />
        </div>
      </div>
      <div
        className="justify-content-md-center row"
      >
        <div
          className="col-md-8 col-12"
        >
          <form />
        </div>
      </div>
    </div>
  </section>,
]
`;

exports[`ContactPageTemplate renders with html content 1`] = `
[
  <header />,
  <section
    className="section"
  >
    <div
      className="container"
    >
      <div
        className="justify-content-md-center row"
      >
        <div
          className="col-md-8 col-12"
        >
          <div
            dangerouslySetInnerHTML={
              {
                "__html": <p>
                  HTML content
                </p>,
              }
            }
          />
          <hr />
        </div>
      </div>
      <div
        className="justify-content-md-center row"
      >
        <div
          className="col-md-8 col-12"
        >
          <form />
        </div>
      </div>
    </div>
  </section>,
]
`;

exports[`ContactPageTemplate renders without content 1`] = `
[
  <header />,
  <section
    className="section"
  >
    <div
      className="container"
    >
      <div
        className="justify-content-md-center row"
      >
        <div
          className="col-md-8 col-12"
        >
          <form />
        </div>
      </div>
    </div>
  </section>,
]
`;
